titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/v0/V0SpecPodFactory.java

```diff
@@ -177,6 +183,39 @@ public V1Pod buildV1Pod(Job<?> job, Task task, boolean useKubeScheduler, boolean
         return new V1Pod().metadata(metadata).spec(spec);
     }
 
+    private List<V1Volume> buildV1Volumes(List<Volume> volumes) {
+        if (volumes == null) {
+            return Collections.emptyList();
+        }
+        List<V1Volume> v1Volumes = new ArrayList<>();
+        for (Volume v : volumes) {
+            v1Volumes.add(buildV1Volume(v));
+        }
+        return v1Volumes;
+    }
+
+    private V1Volume buildV1Volume(Volume volume) {
+        if (volume.getVolumeSource() instanceof SharedContainerVolumeSource) {
+            V1FlexVolumeSource flexVolume = getV1FlexVolumeForSharedContainerVolumeSource(volume);
+            return new V1Volume()
+                    .name(volume.getName())
+                    .flexVolume(flexVolume);
+        } else {
+            // SharedVolumeSource is currently the only supported volume type
+            return null;
```

Comment:
I think this should never be null. Either we have `Volume` with valid data or the `Volume` list should be empty.

Reply:
I think I (finally) understand what you are saying. 
You are (rightfully) concerned we might accidentally have this in a pod spec:

```
volumes: [ {volume1}, null ],
```

I think the Optional thing sounds like a good choice here, and simply not return anything and then have the caller not add anything to `volumes` in that case.